File: LC/Python/Easy/1200.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        min_element = min(arr)
        max_element = max(arr)
        shift = -min_element
        sort = [0] * (max_element - min_element + 1)
        
        for num in arr:
            sort[num + shift] = 1
            
        min_diff = max_element - min_element
        prev = 0
        ans = []
        
        for i in range(1, max_element + shift + 1):
            if sort[i] == 0:
                continue
            
            diff = i - prev
            
            if diff < min_diff:
                min_diff = diff
                ans = []
            
            if diff == min_diff:
                ans.append([prev - shift, i - shift])
            prev = i
        
        return ans

File: LC/C++/Easy/1200.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        std::vector<std::vector<int>> ans{};
        int minDiff = std::numeric_limits<int>::max();

        std::sort(arr.begin(), arr.end());

        for (int i = 0; i < arr.size() - 1; ++i) {
            if (std::abs(arr[i + 1] - arr[i]) < minDiff) {
                ans = {{arr[i], arr[i+1]}};
                minDiff = std::abs(arr[i + 1] - arr[i]);
            } else if (std::abs(arr[i + 1] - arr[i]) == minDiff) {
                ans.push_back({arr[i], arr[i+1]});
            }
        }

        return ans;
    }
};